env0-resources-generator/terraform_templates/environments.tftpl

```diff
@@ -7,18 +7,31 @@ resource "env0_environment" "${workspace.name}" {
     %{ if workspace.type == "terraform" ~}
     is_remote_backend = true
     %{ endif ~}
-    without_template_settings {
-        repository = "https://github.com/${workspace.vcs.account}/${workspace.vcs.repository}"
-        path = "${workspace.vcs.project_root}"
-        revision = "${workspace.vcs.branch}"
-        type = "${workspace.type}"
-        %{ if workspace.type == "terraform" ~}
-        terraform_version = "${workspace.terraform_version}"
-        %{ endif ~}
-        %{ if workspace.type == "opentofu" ~}
-        opentofu_version = "${workspace.opentofu_version}"
-        %{ endif ~}
+    %{ if include_vcs_connection ~}
+    dynamic "without_template_settings" {
```

Comment:
dynamic block is the only way to omit null values. it is required here because out provider asserts that only one of the vcs token attribuites is present